openmc/tallies.py

```diff
@@ -1482,19 +1482,31 @@ def _outer_product(self, other, binary_op):
             new_name = '({0} {1} {2})'.format(self.name, binary_op, other.name)
             new_tally.name = new_name
 
+        # Create copies of self and other tallies to rearrange for tally
+        # arithmetic
+        self_copy = copy.deepcopy(self)
+        other_copy = copy.deepcopy(other)
+
         # Find any shared filters between the two tallies
-        self_filters = set(self.filters)
-        other_filters = set(other.filters)
-        filter_intersect = self_filters.intersection(other_filters)
+        filter_intersect = []
+        for filter in self_copy.filters:
+            if filter in other_copy.filters:
+                filter_intersect.append(filter)
```

Comment:
Cool, yeah after reviewing the code that was the only section that popped out at me as a possible culprit. Glad it did the trick!